sys/net/gnrc/sock/udp/gnrc_sock_udp.c

```diff
@@ -166,23 +212,29 @@ ssize_t sock_udp_send(sock_udp_t *sock, const void *data, size_t len,
     gnrc_pktsnip_t *payload, *pkt;
     uint16_t src_port = 0, dst_port;
     sock_ip_ep_t local;
-    sock_ip_ep_t rem;
+    sock_ip_ep_t *rem;
 
     assert((sock != NULL) || (remote != NULL));
     assert((len == 0) || (data != NULL)); /* (len != 0) => (data != NULL) */
-    if ((remote != NULL) && (sock != NULL) &&
-        (sock->local.netif != SOCK_ADDR_ANY_NETIF) &&
-        (remote->netif != SOCK_ADDR_ANY_NETIF) &&
-        (sock->local.netif != remote->netif)) {
-        return -EINVAL;
-    }
-    if ((remote != NULL) && ((remote->port == 0) ||
-                             gnrc_ep_addr_any((const sock_ip_ep_t *)remote))) {
-        return -EINVAL;
+
+    if (remote != NULL) {
+        if (remote->port == 0) {
+            return -EINVAL;
+        }
+        else if (gnrc_ep_addr_any((const sock_ip_ep_t *)remote)) {
+            return -EINVAL;
+        }
+        else if (gnrc_af_not_supported(remote->family)) {
+            return -EAFNOSUPPORT;
+        }
+        else if ((sock != NULL) &&
+            (sock->local.netif != SOCK_ADDR_ANY_NETIF) &&
+            (remote->netif != SOCK_ADDR_ANY_NETIF) &&
+            (sock->local.netif != remote->netif)) {
+                return -EINVAL;
+        }
     }
-    if ((remote == NULL) &&
-        /* sock can't be NULL as per assertion above */
-        (sock->remote.family == AF_UNSPEC)) {
+    else if (sock->remote.family == AF_UNSPEC) {
```

Comment:
Nope, because of the `assert((sock != NULL) || (remote != NULL))`, i.e., this `else if` only fires if `remote == NULL` and per `assert` then `sock != NULL`.